### PR TITLE
New principle: Identity on the web

### PR DESCRIPTION
**Pull request by rhiaro**
***
<a href="https://pr-preview.s3.amazonaws.com/w3ctag/design-principles/pull/396.html" title="Last updated on Mar 6, 2024, 8:48 AM UTC (b4edcf1)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3ctag/design-principles/396/2aa21c4...b4edcf1.html" title="Last updated on Mar 6, 2024, 8:48 AM UTC (b4edcf1)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -286,6 +286,47 @@ Parts of the web platform evolve independently.
...
+* Credentials (data)
+* Authentication & Authorisation (protocols)
+
+When designing, implementing or integrating an "identity" layer into a system,
+you should consider the following:
+
+* Avoid mixing metadata for a system or protocol with the data connected to an identifier for a person.
+* Before you collect data about someone - passively or actively - and attach the data to a persistent identifier, enable them to [give appropriate consent](#consent).
+* Be transparent about the circumstances under which such data is used or shared.
+* The same data is not always appropriate for different scenarios, or in different contexts, so [do not collect more than needed](https://w3ctag.github.io/privacy-principles/#identity) for the service being provided, nor conflate or connect data collected in different situations.
```
Comment (rhiaro) [inline index.bs:321]: TODO: link this properly as a reference

**Comment by rhiaro** [comment]
We talked about this in Breakout C today, and concluded it's long. We can shorten it by taking care to constrain the scope to architectural things, and focus on what really does and doesn't belong in the Design Principles document. Feedback from @LeaVerou and @cynthia would be particularly helpful on this front.

File: index.bs [hunk 2]
```diff
@@ -234,6 +234,46 @@ See also:
...
```
[the remark this answers is not in the document]
Comment (martinthomson) [reply index.bs:268]: I think that you mean https://w3ctag.github.io/privacy-principles/#data-minimization

But then, I don't know who is collecting in this context.  Presumably the designers of Web APIs.  But they don't collect data, they are more often involved in the supply of data.

**Review by martinthomson — COMMENTED** [review]
It's possible that the framing of this as an "identity layer" is the problem.  The concept of identity in the privacy principles is a good one, as it is the all-encompassing notion of how a user presents in a given context.  It doesn't really get down to the level of detail that this appears to want to.

A more useful framing might be the use of identifiers and the data that is then associated with identifiers.  Identifiers as a handle for identity are the locus of most privacy problems as they concretely engage with recognition (all forms: same-context and cross-context; and for any subjective understanding of that distinction).

Of course, the problem with this is that identity is a broad concept.  It might be that the advice here doesn't really belong in a single principle at this level.  The [ethical web principles item on control](https://www.w3.org/TR/ethical-web-principles/#control), that is referenced in the preceding section, is a better expression of the underlying principles.

File: index.bs [hunk 3]
```diff
@@ -286,6 +326,7 @@ Parts of the web platform evolve independently.
 Issues that are present with a certain web technology now may be fixed in a subsequent iteration.
 Duplicating these issues makes fixing them more difficult.
 By adhering to this principle we can make sure overall platform quality improves over time.
+
```
Comment (cynthia) [inline index.bs:329]:
```suggestion
```

**Review by martinthomson — COMMENTED** [review]
A missing point that occurs to me is that identifiers will often be attached to user agents (or really how that user agent presents itself to a site: think IP address or state partitioning) as a proxy for a person.  In effect, the distinction between a person and how they present themselves can be important, but it is often conflated.  So we need to apply the same protections to these proxies as we would the people they model.

File: index.bs [hunk 4]
```diff
@@ -236,6 +236,38 @@ See also:
...
+Features that use or depend on
+identifiers and the attachment of data about a person
+to that identifier carry privacy risks which
+often reach beyond a single API or system.
+This includes data that has been passively generated (for example,
+about their behaviour on the web) as well as that which has been
+actively collected (for example, they have filled in a form).
+
+For such features, you should [understand the context](https://w3ctag.github.io/privacy-principles/#identity)
```
Comment (torgo) [inline index.bs:261]:
```suggestion
For such features, you should [understand the context]([[PRIVACY-PRINCIPLES#identity]])
```

File: index.bs [hunk 5]
```diff
@@ -236,6 +236,38 @@ See also:
...
+in which it will be used,
+including how it will be used alongside other features of the web.
+Make sure the user can [give appropriate consent](#consent).
+Design APIs to collect
+[the smallest amount of data](https://w3ctag.github.io/privacy-principles/#data-minimization)
```
Comment (torgo) [inline index.bs:266]:
```suggestion
[the smallest amount of data]([[PRIVACY-PRINCIPLES#data-minimization]])
```

File: index.bs (hunk 4 above)
Comment (rhiaro) [inline index.bs:261]:
```suggestion
For such features, you should [understand the context](https://www.w3.org/TR/privacy-principles/#identity)
```

File: index.bs (hunk 5 above)
Comment (rhiaro) [inline index.bs:266]:
```suggestion
[the smallest amount of data](https://www.w3.org/TR/privacy-principles/#data-minimization)
```

File: index.bs [hunk 6]
```diff
@@ -236,6 +236,38 @@ See also:
 
 * [Security and privacy are essential](https://www.w3.org/2001/tag/doc/ethical-web-principles/#privacy)
 
+<h3 id="identity">Use identity appropriately in context</h3>
+
+Give people [control](https://www.w3.org/TR/ethical-web-principles/#control)
+over the identifying information about themselves
+they are presenting in different contexts on the web,
+and be transparent about it.
```
Comment (cynthia) [inline index.bs:244]: Active tone would be preferred, along with brevity.

File: index.bs [hunk 7]
```diff
@@ -294,6 +326,7 @@ Parts of the web platform evolve independently.
 Issues that are present with a certain web technology now may be fixed in a subsequent iteration.
 Duplicating these issues makes fixing them more difficult.
 By adhering to this principle we can make sure overall platform quality improves over time.
+
```
Comment (cynthia) [inline index.bs:329]:
```suggestion
```

**Review by cynthia — CHANGES_REQUESTED** [review]

**Review by cynthia — APPROVED** [review]

**Review by torgo — APPROVED** [review]

**Review by martinthomson — COMMENTED** [review]
A belated approval from me.

**Pull request merged**